synapse/storage/databases/main/events.py

```diff
@@ -1172,9 +1172,15 @@ def _update_current_state_txn(
             # Invalidate the various caches
 
             for member in members_changed:
+                # We need to invalidate the cache for get_rooms_for_user_with_stream_ordering
+                # both with and without excluding rooms for sync results.
                 txn.call_after(
                     self.store.get_rooms_for_user_with_stream_ordering.invalidate,
-                    (member,),
+                    (member, False),
+                )
+                txn.call_after(
+                    self.store.get_rooms_for_user_with_stream_ordering.invalidate,
+                    (member, True),
                 )
```

Comment:
This looks a bit meh to me but I couldn't find out another way to invalidate the cache for `member` regardless of the value of the second parameter.